dvc/repo/fetch.py

```diff
@@ -143,21 +143,28 @@ def _fetch(repo, obj_ids, **kwargs):
 
 
 def _fetch_partial_imports(repo, targets, **kwargs):
+    from dvc.dependency.repo import RepoDependency
     from dvc.stage.exceptions import DataSourceChanged
 
     downloaded = 0
     failed = 0
+    jobs = kwargs.pop("jobs", None)
     for stage in repo.partial_imports(targets, **kwargs):
         dep = stage.deps[0]
         out = stage.outs[0]
+
         try:
             if dep.changed_checksum():
                 raise DataSourceChanged(f"{stage} ({dep})")
 
             logger.info("Importing '%s' -> '%s'", dep, out)
-            out.transfer(
-                dep.def_path, jobs=kwargs.get("jobs"), odb=repo.odb.local
-            )
+            if isinstance(dep, RepoDependency):
+                obj = dep.transfer(repo.odb.local, jobs=jobs)
+                out.hash_info = obj.hash_info
+                out.meta = dep.get_meta()
```

Comment:
This behavior seems incorrect for repo dependencies. We should not need to use `transfer` at all for repo dependencies, since we can already use the regular `fetch` behavior to download them (by collecting the used objects from the source repo.

Reply:
Also, maybe I'm misunderstanding something, but even in the partial import case, these objects should have already been collected and fetched in the regular `_fetch()` step (since even when out has no hash_info, the used obj collection will still return the repo dependency's used objects).

So it looks like maybe all you need to do for partial repo imports is to update the output hash info so it matches `dep.get_obj().hash_info`?

Reply:
Thanks @pmrowla. Updated as discussed.